test/h3core.spec.js

```diff
@@ -287,14 +286,13 @@ test('kRing 2', assert => {
     assert.end();
 });
 
-test('kRing - Bad Radius', assert => {
-    const hexagons = h3.kRing('8928308280fffff', -7);
-    assert.deepEqual(hexagons, ['8928308280fffff'], 'Got origin for bad radius');
+test('gridDisk - Bad Radius', assert => {
+    assert.throws(() => h3.gridDisk('8928308280fffff', -7), /\(2\)$/, 'Throws with bad radius');
```

Comment:
I see what you're saying about making the error codes machine readable - even in test it would be really nice to have a better way to assert which error we got.

One option here is to make subclasses for each error:
```
class H3ErrorFailed extends Error {
  constructor() {
    super('The operation failed but a more specific error is not available');
    this.name = 'H3ErrorFailed';
  }
}
```
Then we can either check `error.name` or `error instanceof H3ErrorFailed` to check the error type. I'd need to check that this works in all compile targets, though, and it might be best for a separate PR